www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+hops. For this version of the protocol, we propose upfront fees as a credential acquisition
+method. Payment HTLC are sent to the routing hop and credentials are returned back to the sender
+with a onion reply path. The credentials are counter-signed by the issuer to authenticate their
+origin. The amount of satoshis and the corresponding quantity of credentials delivered is defined
+by each routing hop as a routing policy decision, there is no global pricing of the channel liquidity.
+Alternative acquisiton methods (e.g [proof-of-UTXO ownership](https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-November/002884.html))
+can be experimented and deployed in a compatible fashion, without network-wide coordination.
+
+Once a HTLC sender has bootstrapped the possession of sufficient credentials to route across
+the network to reach its payee destination, payment paths can be drawn. For each hop in the path,
+the HTLC onion payload should contain the correct amount of credentials, as requested by the
+hop routing policy.
+
+At reception of a HTLC forward, the credentials signatures should be verified and enforced against
+the routing hop policy. If the HTLC sender has staked enough credentials to lockup outbound liquidity
+for the request msat amount and CTLV duration, the HTLC forward should be committed on the outbound
+channel, and the HTLC setup phase pursued until the destination payee is reached.
+
+At HTLC settlement, if the HTLC is successful, the routing fees are paid accordingly to current
+`channel_update`, a new set of credentials is issued and counter-signed by the hop and returned back
+to the sender through [onion communication channels](https://github.com/lightning/bolts/pull/759).
+A supplement of credentials can be joined, as reward of honoring the routing fees, and unlocking more
+consequential liquidity lockups for future HTLC forwards.
+
+If the HTLC is a failure, the reputation credentials are slashed. No new credentials are issued back
+to the HTLC sender as a punishment. To limit reputation whitewashing, where the jamming damage is superior
+to the reputation acquisition cost, a proportion sould be maintained between the credentials acquisition
+cost (either from upfront fees or honored routing fees) and the channel liquidity lockup rights.
+
+To preserve the confidentiality of the HTLC senders (or payers), the credentials should be anonymized,
+a routing hop should not be able to link between a credential issuance at dissemination and its usage
+at the reception of a HTLC forward request. A cryposystem satisfying those requirements could be to
+leverage EC [blinded signatures](https://sceweb.sce.uhcl.edu/yang/teaching/csci5234WebSecurityFall2011/Chaum-blind-signatures.PDF).
+
+Additionally, this proposal is extending the `channel_update` message to announce routing fees paid on the
+CLTV duration, therefore allowing compensation of the routing hops for the transport of long-term held
```

Comment:
Hold invoice may have a longer cltv, but I don't think that that is a suitable way to recognize and pay for them. Also quick-resolving htlcs on longer routes may have high cltv deltas. What makes hold invoices costly is their actual hold time.

Reply:
I think effectively we don't have a suitable, "quantifiable" way to dissociate between short-term held HTLCs and long-term held HTLCs, other than an arbitrary protocol/implementation-defined value. On the LDK-side, we reject a any outgoing HTLC with a CLTV superior to 2016 blocks from our tip. I don't think it's even opportune to introduce a "discrete" approach but rather design an anti-jamming solution covering both "continuously". 

Making the HTLC upfront fees/reputation cost actually paid on the "actual hold time" would be more valuable. A HTLC sender would only have to cover cost for effective duration, not worst-case CLTV scenario. I think this could be possible if `update_fulfill_htlc` paid back routing fee scaled on a new `fee_base_block` to encompass the time dimension.

Reply:
Yes, or in the credentials case let the routing node sign a number of credentials that varies based on the actual hold time? If the sender causes a lot of holding, they will keep sending out more credentials than they get back, bringing their credentials balance down. The onion payload is limited in size though, so it seems that it can't be very granular.

Reply:
If you assume there is a satoshi-cost behind each credential unit, I think you should scale the quantity of credentials requested on the outbound CLTV duration as expressed in the Script during the forward phase. There is no a priori knowledge of the HTLC actual hold time.

Then at settlement, if the observed actual hold time is above some average/median, the routing node can return less credentials, leading the sender to pre-pay for more credentials if it would like to keep doing a lot of holding.

I think this alternative also work, rather than paying routing fee scaled on the time dimension, however I think this more risky for the routing node, as the actual holding time of a HTLC might be far above your median. Requesting credentials to cover the worst-scenario is zero-risk for the routing node, at the trade-off of more expensive for the HTLC senders.

About the granularity, if the onion communication channel are detached from the payment paths, you could send multiple onions, overreaching beyond the 32kb limit. The granularity wished between msat/sat and credentials is a good question.

Reply:
Yes, the difference in hold time between a regular payment and a hold payment is massive. Hard to bridge that gap using credentials without requiring more trust.

Routing fee scaling on a time dimension - I think it only works for the success case? Seems we also need something for the slow-failed case.

Reply:
For the slow-failed case, under the credentials paradigm, a routing node would have to scale the quantity of credentials requested to accept the HTLC forward. E.g, if the HTLC sender requests a 2-month outgoing CLTV, a routing node policy should be to have previously satisfied enough 4*2016 block period (2016 max standard allowed outgoing CLTV). The HTLC sender should have either "harvest" the credentials from previous satisfied routing success, or pre-paid 4 block period.